virtual-keyboard/script.js

```diff
@@ -0,0 +1,139 @@
+
+        this.elements.main.append(this.elements.keysContainer);
+        document.body.append(this.elements.main);
+
+        document.querySelectorAll('.use-keyboard-input').forEach(element => {
+            element.addEventListener('focus', () => {
+                this.open(element.value, currentValue => {
+                    element.value = currentValue;
+                });
+            });
+        });
+    },
+
+    _createKeys() {
+        const fragment = document.createDocumentFragment();
+        const keyLayout = [
+            "1", "2", "3", "4", "5", "6", "7", "8", "9", "0", "backspace",
+            "q", "w", "e", "r", "t", "y", "u", "i", "o", "p",
+            "caps", "a", "s", "d", "f", "g", "h", "j", "k", "l", "enter",
+            "done", "z", "x", "c", "v", "b", "n", "m", ",", ".", "?",
+            "space"
+        ];
+        const createIconHTML = (icon_name) => `<i class=material-icons>${icon_name}</i>`;
+        
+        keyLayout.forEach(key => {
+            const keyElement = document.createElement('button');
+            const insertLineBreak = ['backspace', 'p', 'enter', '?'].indexOf(key) !== -1;
+        
+            keyElement.setAttribute('type', 'button');
+            keyElement.classList.add('keyboard__key');
+
+            if (key === 'backspace') {
+                keyElement.classList.add('keyboard__key_wide');
+                keyElement.innerHTML = createIconHTML('backspace');
+                keyElement.addEventListener('click', () => {
+                    this.properties.value = this.properties.value.substring(0, this.properties.value.length-1);
+                    this._triggerEvent('oninput');
+                });
+            } else if (key === 'caps') {
+                keyElement.classList.add('keyboard__key_wide', 'keyboard__key_activatable');
+                keyElement.innerHTML = createIconHTML('keyboard_capslock');
+                keyElement.addEventListener('click', () => {
+                    this._toggleCapsLock();
+                    keyElement.classList.toggle('keyboard__key_active', this.properties.capsLock);
+                });
+            } else if (key === 'enter') {
+                keyElement.classList.add('keyboard__key_wide');
+                keyElement.innerHTML = createIconHTML('keyboard_return');
+                keyElement.addEventListener('click', () => {
+                    this.properties.value += '\n';
+                    this._triggerEvent('oninput');
+                });
+            } else if (key === 'space') {
+                keyElement.classList.add('keyboard__key_extra-wide');
+                keyElement.innerHTML = createIconHTML('space_bar');
+                keyElement.addEventListener('click', () => {
+                    this.properties.value += ' ';
+                    this._triggerEvent('oninput');
+                });
+            } else if (key === 'done') {
```

Comment:
Hi @andreiha. You use a lot of else if. So, in this cause try to use switch/case